web/controllers/auth.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Dwylbot.Auth do
+  import Plug.Conn
+  import Phoenix.Controller
+  alias Dwylbot.Router.Helpers
+
+  def init(_opts) do
+  end
+
+  def call(conn, _opts) do
+    # this OR operator is used for our tests
+    # the test assign values directly to the conn instead of creating some fake sessions/cookies
+    # see Chapter 8 page 137 section Integration Tests of the Programmin Phoenix book
+    user = get_session(conn, :current_user) || conn.assigns[:current_user]
```

Comment:
Pg 137 doesn't seem to do this. Is this the right bit of code? It seems a bit different
![screen shot 2017-05-08 at 22 17 56](https://cloud.githubusercontent.com/assets/15983736/25825817/51c8dc96-343c-11e7-98b9-3b0243c74b63.png)
Can't find examples of `get_session(conn, :current_user)`, just `get_session(conn, :user_id)`